src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -262,15 +262,21 @@ impl VfioDevice {
         let dev_type = dev_info.dev_type.as_str();
         let driver_type = VfioBusMode::driver_type(dev_type).to_owned();
 
-        Self {
+        let mut vfio_device = Self {
             device_id,
             attach_count: 0,
             bus_mode: VfioBusMode::PCI,
             driver_type,
             config: dev_info.clone(),
             devices,
             device_options,
-        }
+        };
+
+        vfio_device
+            .initialize_vfio_device()
+            .context("initialize vfio device failed.")?;
+
```

Comment:
Not only the vfio device setup code is moved to a separate function, but it is now called at device creation time, not at device attach time. Can this change behavior or expectations of some other code ? In any case, this should be highlighted in the commit change log.

Reply:
Thx @gkurz I will add info to the commit message to highlight that this change does not change the behavior of other code and is in line with expectations.

Reply:
fixed